drivers/net/dsa/lantiq_gsw_core.c

```diff
@@ -363,40 +118,42 @@ static void gswip_switch_mask(struct gswip_priv *priv, u32 clear, u32 set,
 static u32 gswip_switch_r_timeout(struct gswip_priv *priv, u32 offset,
```

Comment:
The body of this function is currently commented out as I'll need to get far enough along to learn whether I can drop the `gswip_priv` argument from the `ops.read()` call. If I can, it will just call `priv->ops->read()` and stay here. If not, I will likely convert it into a bus-specific function and add it to the `ops` struct.

Reply:
The `readx_poll_timeout` macro expects a read function prototype with only a single argument (currently the offset). In the MDIO case there is additional needed information: a handle to the MDIO device in order to call the correct bus function, know what address on the bus to target, etc.

There are a couple options: move this function to the bus-specific files, bundle the necessary arguments into a struct and pass that, or make the MDIO handle / state info global and remove the need to pass it at all. Not sure exactly which approach I'll take yet...

Reply:
I am not a fan of the global option I think the other option sounds a lot better

Reply:
> I am not a fan of the global option I think the other option sounds a lot better. You could also just add a function pointer for the timeout read variant? So you'd have `priv->ops->read(address)` and `priv->ops->r_timeout(address)`

Ended up going that route, yup. See https://github.com/bolt-snap-bolt/linux/compare/feature/WARP-5838...bolt-snap-bolt:feature/WARP-5840